Budgeting and goals

23722943 opened the pull request:
Budgeting

Current features:
Create a simple budget with:
→ {category, limit, period}
Display budgets in a table

Planned features to add:

Link to transactions so spending updates automatically
Add a bar showing date/time left in the budget period
Display the amount already spent in each category

Goals

Current features:
Create a simple goal with:
→ {title, current amount, target amount, deadline}
Display goals in a table
Show progress with a progress bar
Display time left in the table

Planned features to add:
Link to transactions so progress updates automatically
Add a feature to tick off completed goals, grey them out, and move them to the bottom of the table

23722943 committed:
budgeting_and_goals

23722943 committed:
Added Progress Bar and Time Left for Goals

23722943 committed:
Used Bootstrap and Added Comments

23722943 committed:
Fixed the functionallity for the delete button for budgeting and goals

sepehrmoghani commented:
looks good. please pull the main branch. that one contains all the work we've done so far so it will include other apps as well.
planned features look good. 
if you like to make any changes or modifications to the front end, please feel free to do so. but please ensure you work upon the work that's already been done. thanks

23722943 committed:
Added checks to date selection and added field to the goals form 'start_date'; Category is now a selector field and the user must select one of the categories within the selector

23722943 committed:
Updated to include the files from main

Added checks to date selection and added field to the goals form 'start_date'; Category is now a selector field and the user must select one of the categories within the selector

23722943 committed:
Refactor budgeting and goals forms and routes; add transaction retrieval logic and update templates for budget summaries

23722943 committed:
Update goals can be updated with transactions

23722943 committed:
Add privacy setting to goals db

23722943 committed:
Add description fields to budget and goal forms; enhance goal statistics and shared goals view

23722943 committed:
Add like and share button functionality
- Implemented like/save toggles with AJAX
- Created new explore page (to share goals)

Add 4 new widgets to budget dashboard

23722943 committed:
Small widget icon changes

23722943 committed:
merge to main

23722943 committed:
Merge branch 'main' into budgeting_and_goals
